internal/commands/app/describe_test.go

```diff
@@ -0,0 +1,380 @@
+				c.Send("test-app-abcd")
+				c.SendLine(" ")
+				c.ExpectEOF()
+			},
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			_, console, _, ui, err := mock.NewVT10XConsole()
+			assert.Nil(t, err)
+			defer console.Close()
+
+			doneCh := make(chan (struct{}))
+			go func() {
+				defer close(doneCh)
+				tc.procedure(console)
+			}()
+
+			atlasClient := mock.AtlasClient{}
+			atlasClient.GroupsFn = func() ([]atlas.Group, error) {
+				return tc.groups, nil
+			}
+
+			realmClient := mock.RealmClient{}
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				return tc.apps, nil
+			}
+			var groupIDActual, appIDActual string
+			realmClient.AppDescriptionFn = func(groupID, appID string) (realm.AppDescription, error) {
+				groupIDActual = groupID
+				appIDActual = appID
+				return realm.AppDescription{}, nil
+			}
+
+			cmd := &CommandDescribe{inputs: tc.inputs}
+			assert.Nil(t, cmd.Handler(nil, ui, cli.Clients{Atlas: atlasClient, Realm: realmClient}))
+			assert.Equal(t, "123", groupIDActual)
+			assert.Equal(t, "456", appIDActual)
+		})
+	}
+
+	t.Run("should describe app", func(t *testing.T) {
+		out, ui := mock.NewUI()
+
+		realmClient := mock.RealmClient{}
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{{ID: "456", ClientAppID: "test-app-abcd", GroupID: "123"}}, nil
+		}
+		realmClient.AppDescriptionFn = func(groupID, appID string) (realm.AppDescription, error) {
+			return realm.AppDescription{
+				ClientAppID: "todo-abcde",
+				Name:        "todo",
+				RealmURL:    fmt.Sprintf("https://admin-base.url/groups/%s/apps/%s/dashboard", groupID, appID),
+				DataSources: []realm.DataSourceSummary{
+					{
+						Name:       "mongodb-atlas",
+						Type:       "mongodb-atlas",
+						DataSource: "Cluster0",
+					},
+					{
+						Name:       "mongodb-datalake",
+						Type:       "datalake",
+						DataSource: "DataLake0",
+					},
+					{
+						Name: "mdb1",
+						Type: "mongodb",
+					},
+				},
+				HTTPEndpoints: []realm.HTTPEndpointSummary{
+					{
+						Name: "http",
+					},
+				},
+				ServiceDescs: []realm.ServiceSummary{
+					{
+						Name: "tw1",
+						Type: "twilio",
+					},
+				},
+				AuthProviders: []realm.AuthProviderSummary{
+					{
+						Name:    "oauth2/google",
+						Type:    "oauth2/google",
+						Enabled: true,
+					},
+					{
+						Name:    "oauth2/facebook",
+						Type:    "oauth2/facebook",
+						Enabled: true,
+					},
+				},
+				CustomUserData: realm.CustomUserDataSummary{
+					Enabled:     true,
+					DataSource:  "Cluster0",
+					Database:    "db1",
+					Collection:  "col1",
+					UserIDField: "id",
+				},
+				Values: []string{
+					"value1",
+					"value2",
+				},
+				Hosting: realm.HostingSummary{
+					Enabled: true,
+					Status:  "setup_ok",
+					URL:     "https://hosting.domain/",
+				},
+				Functions: []realm.FunctionSummary{
+					{
+						Name: "func1",
+						Path: "func1.js",
```

Comment:
unrelated to the test functionality, but `path` here is intended to be blank, and below it'd just be `nested`